packages/framework-provider-aws/test/library/events-adapter.test.ts

```diff
@@ -131,8 +132,75 @@ describe('the events-adapter', () => {
       )
     })
   })
+
+  describe('the `destroyEntity` method', () => {
+    it('deletes all entity events and snapshots', async () => {
+      const entityID = random.uuid()
+      const entityName = random.word()
+      const numberOfSnapshots = random.number({ min: 1, max: 10 })
+      const numberOfEvents = random.number({ min: 1, max: 10 })
+      const fakeDelete = fake.returns({
+        promise: fake.resolves(''),
+      })
+      const fakeQuery = stub()
+        .onFirstCall()
+        .returns({
+          promise: fake.resolves({
+            Items: buildRandomEventEnvelopesForEntity(numberOfEvents, entityName, 'event', entityID),
+          }),
+        })
+
+      fakeQuery.onSecondCall().returns({
+        promise: fake.resolves({
+          Items: buildRandomEventEnvelopesForEntity(numberOfSnapshots, entityName, 'snapshot', entityID),
+        }),
+      })
+
+      const fakeBatchWrite = fake.returns({
+        promise: fake.resolves(''),
+      })
+      const dynamoDB: DocumentClient = { delete: fakeDelete, query: fakeQuery, batchWrite: fakeBatchWrite } as any
+      const config = new BoosterConfig('test')
+      config.appName = 'nuke-button'
+
+      await Library.destroyEntity(dynamoDB, config, fakeLogger, entityName, entityID)
+
+      expect(fakeBatchWrite).to.be.calledWith(
+        match({
+          RequestItems: match.has(
+            config.resourceNames.eventsStore,
+            match.has('length', numberOfSnapshots + numberOfEvents)
+          ),
+        })
+      )
+    })
+  })
 })
 
+function buildRandomEventEnvelopesForEntity(
+  numberOfEvents: number,
+  entityName: string,
+  kind: 'event' | 'snapshot',
+  entityId: string
+): Array<EventEnvelope> {
+  const events = new Array<EventEnvelope>(numberOfEvents)
+  for (let i = 0; i < numberOfEvents; i++) {
+    events[i] = {
+      version: 1,
+      entityID: entityId,
+      kind: kind,
+      value: {
+        id: entityId,
+      },
+      typeName: random.word(),
+      entityTypeName: entityName,
+      requestID: random.uuid(),
+      createdAt: date.past().toISOString(),
+    }
+  }
+  return events
```

Comment:
Perhaps this, in order to avoid future off-by-one errors?

```suggestion
  return new Array<EventEnvelope>(numberOfEvents)
    .fill(undefined)
    .map(() => ({
      version: 1,
      entityID: entityId,
      kind: kind,
      value: {
        id: entityId,
      },
      typeName: random.word(),
      entityTypeName: entityName,
      requestID: random.uuid(),
      createdAt: date.past().toISOString(),
    }))
```